### PR TITLE
Adding new tutorial on the impact of grid and velocity interpolation scheme on trajectories

### PR DESCRIPTION
**Pull request by erikvansebille**
This new tutorial was developed with @sruehs and @michaeldenes during discussions on the relation between grids and interpolation schemes.

### CODE REVIEW

**Review by VeckoTheGecko — CHANGES_REQUESTED** [review]
I have made some changes. Rather than pushing, I'd like to discuss first
[suggested_diff.html.zip](https://github.com/user-attachments/files/17896608/suggested_diff.html.zip)

Changes:
- Introduce utility class to consolidate experiment handling (getting interpolation method, advection kernel, file name, plot title)
- patch fstring to be 3.10 compatible
- rename matplotlib variables to align with convention
- update plotting code to manually handle legend

Ignore the `sharex=` in the plt.subplots code (will remove that, its not needed).

My only concern is in the last plot. My changes caused a small change in the red dots "Cgrid & analytical advection". Any idea why this would be the case? I can hunt further

**Review by VeckoTheGecko — APPROVED** [review]

**Comment by erikvansebille** [comment]
Thanks for this clean-up, @VeckoTheGecko. Defining a class is perhaps a bit overkill for a tutorial (we don't do it in other tutorials) but on the other hand can also inspire Parcels users (including me) to work more with classes themselves

Note that there was a small change in your last figure, where the red dots ('Cgrid and analytical advection' were different in your rendering. If I rerun your code, I get the original values for these experiments. Not sure why you get (slightly) different results for analytical advection; perhaps the version of scipy?? Anyways, since we're working on a newer, more robust implementation of AnalyticalAdvection anyways (see #1612), I don't worry too much for now

**Comment by erikvansebille** [comment]
@michaeldenes or @sruehs, do you still have comments/feedback on this tutorial before I merge into master? We can of course still update/amend it after merging too

**Comment by VeckoTheGecko** [comment]
> Note that there was a small change in your last figure,

Yes, I noticed that as well. I agree that its likely dependencies 

> Defining a class is perhaps a bit overkill for a tutorial

Yes, that's fair. If it hinders understanding I'm happy to revert (I'm not too committed to the class approach - I just find it more readable but I'm also not the average Parcels user, so good to know what others think :) )

File: docs/examples/tutorial_peninsula_AvsCgrid.ipynb [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+   "metadata": {},
+   "source": [
+    "# The impact of grid and velocity interpolation scheme on trajectories"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The notebook below shows how sensitive trajectories are to the combination of grid and velocity interpolation schemes. We will use the Peninsula example (also discussed in the [Quickstart to Parcels tutorial](https://docs.oceanparcels.org/en/latest/examples/parcels_tutorial.html#Sampling-a-Field-with-Particles)), but carefully compare how trajectories on the [A-grid and C-grid](https://docs.oceanparcels.org/en/latest/examples/documentation_indexing.html) version of this flow differ. To enhance the differences, we deliberately use a very course-resolution grid."
```
Comment (michaeldenes) [inline docs/examples/tutorial_peninsula_AvsCgrid.ipynb:14]: "but carefully compare" -> "and carefully compare".

"course-resolution" -> "coarse-resolution"

File: docs/examples/tutorial_peninsula_AvsCgrid.ipynb [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+    "from matplotlib.lines import Line2D\n",
+    "\n",
+    "import parcels"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We start with defining the `FieldSet` for the Peninsula flow and the grid. Note that on the A-grid we can directly calculate the velocities on the grid pints from the streamfunction `P`. On the C-grid, we need to differentiate the streamfunction `P` to get the velocities on the staggered `U` and `V` grid points."
```
Comment (michaeldenes) [inline docs/examples/tutorial_peninsula_AvsCgrid.ipynb:39]: "pints" -> "points"

File: docs/examples/tutorial_peninsula_AvsCgrid.ipynb [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
...q5uxA69+D2pWoKDqzuRvfvRNbE33Xr2vSCK81NRnlfL+v8dlpVwvSeGten31aWloc+UFhVOffuiCratZn2gdJ+8nejXU94WKqT0NhHh8lSiAybvcBS5kkJJqMxEDEok3DMCZ5UzGoOmWd8FqTBSk6VL3ZLFpyXRVQ5KRt/eg0Vv70LXYCDnUBk7TklhO59QqlTmcan3BTnfhYZCqaRLQi/C+w38x4hHTY2OlZ/vR9sgkWCdYn0YODvqtIuOhg3rMdVKNmmuo0ajcG7uz55bm43GIKmFYnzi5P8L1RYbE2sirskSwuUSJeEAUKgkW4iaPBs7l0SrIOGW/I3ckXiPTTPXCZOoW7gAkAjQ05HoIC2uR9/eg8Xv7qK2rJHKgnrWfHOQsXcltVhs8nTw7RrKlCdeIufAHrb98YNcGFg0meRaBhv+9zlBkTGE9R1IeN+BeAWFnOZdL06EJw+isb6OiORB+HYJveC2NtoGPSs+24+mRhr7AsI9GXJd3Gmvw9TYSL25iJng5oazlf+qNQ6WWgKx8rywrghBlPqp6GEJtNl4ol+i/dfGE92q7/YM6INSUGIUjazPW8sDvR9t9hsoXFxxGTGS+uXLMFVX07Bpk10LnSa4+Tgx5s5Eln24B4POxMndxbj5ONFvWkT7fzA7UDs5SbWE+g/GoNORe2gvIbEJp294EaBLYm9qcjKJHTaa+BHj8Ao+v6IeMNu4/HREDlQnjOraqt2S3K6hgcZ90lxO1bUr6hDba91dvFPe7h3YsY69UHCfOYu6+fPQHjyA/mQGRbfdQuAXX9otznyhoXZU0bm7r7wGM+iMqC6hZ7KLlzeent44urri4umNq48vngHBeAV3wqdLN9x8/P4xi7m0DXkcWiuJIwSFwMibE05bC65x926MJVKxUJfBg23uoaL6QtmOuEloIdRKJLqoUIPzpS9W+CfhlJiE+5wrqf39N0RNA2XPPE3MN/+ltryRA6uyQYQNPxxBqVYSmmRbd6j78LEERkSz6tN3KM/NwmQ0sHP+rxz/ewMp19xKaO/ki8YKsYNEbwVXvvohgZ1Pb5lyPnFydzHb5p2QXw++OpbA8NNHTkVRpG7JX9ILQcBt4iSb/bW6Gk5WS4RbtHcsTirLQl+Rby4q6uCO6HthSKGLAeGeEbiq3ajX13GwbL9NkTOQVEtN6uG6RQtPS6IDRPUPpiK/j
```
Comment (michaeldenes) [inline docs/examples/tutorial_peninsula_AvsCgrid.ipynb:294]: "Particles" -> "particles"

Comment (michaeldenes) [inline docs/examples/tutorial_peninsula_AvsCgrid.ipynb:407]: "Parcels" -> "`Parcels`".

**Review by michaeldenes — COMMENTED** [review]
I've added some specific comments (via GitHub) to the notebook.

**Comment by michaeldenes** [comment]
Sorry for the delay in review, didn't see this until today!

File: docs/examples/tutorial_peninsula_AvsCgrid.ipynb (hunk 3 above)
Comment (VeckoTheGecko) [reply docs/examples/tutorial_peninsula_AvsCgrid.ipynb:407]: I think its fine having "Parcels" without the code formatting?

Comment (VeckoTheGecko) [reply docs/examples/tutorial_peninsula_AvsCgrid.ipynb:407]: Keep as is for consistency with other tutorials. Micael agrees

**Review by michaeldenes — APPROVED** [review]
Looks good to me now!

**Pull request merged**